api/games/methods.js

```diff
@@ -60,3 +62,58 @@ export const updateGameData = new ValidatedMethod({
     }
   }
 });
+
+export const earlyExitGame = new ValidatedMethod({
+  name: "Games.methods.earlyExitGame",
+
+  validate: new SimpleSchema({
+    gameId: {
+      type: String,
+      regEx: SimpleSchema.RegEx.Id
+    },
+    endReason: {
+      label: "Reason for End",
+      type: String,
+      regEx: /[a-zA-Z0-9_]+/
+    }
+  }).validator(),
+
+  run({ gameId, endReason }) {
+    if (!Meteor.isServer) {
+      return;
```

Comment:
This will ensure we don't _process_ this on the front-end, but it can still be _called_ from the front-end.

I think we should block the front-end from being able to end the game.

I believe you can do this by checking [`this.connection`](https://docs.meteor.com/api/methods.html#DDPCommon-MethodInvocation-connection) in the method body, if the connection is present, it originated from the client, and you throw with a "not allowed".